warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EdgeTableRangeBuildingVisitor.java

```diff
@@ -410,10 +400,17 @@ private void checkNotExclusion(IdentityContext context, String msg) {
      */
     @Override
     public Object visit(ASTEQNode node, Object data) {
-        termCount++;
+        incrementTermCountAndCheck();
         return visitExpresionNode(node, EQUALS);
     }
 
+    private void incrementTermCountAndCheck() {
+        if (++termCount > maxTerms) {
+            log.error("Query has too many terms");
+            throw new IllegalArgumentException("Too many search terms " + termCount);
```

Comment:
Even though this will error out at one over the threshold, I would make this a little more descriptive with something like 
```
"Term count X exceeded term threshold of Y"
```

Reply:
Since we terminate as we go, it'll always just say like `Term count 11 exceeded term threshold of 10` even if the query has 100 terms. I updated the verbiage to hopefully be more understandable. Let me know what you think.